src/Workleap.OpenApi.MSBuild/OasdiffManager.cs

```diff
@@ -1,35 +1,81 @@
+using System.Runtime.InteropServices;
+
 namespace Workleap.OpenApi.MSBuild;
 
 internal sealed class OasdiffManager : IOasdiffManager
 {
     private const string OasdiffVersion = "1.9.2";
 
+    private readonly ILoggerWrapper _loggerWrapper;
     private readonly IHttpClientWrapper _httpClientWrapper;
     private readonly IProcessWrapper _processWrapper;
-    private readonly string _toolDirectory;
+    private readonly string _openApiToolsDirectory;
+    private readonly string _oasdiffDirectory;
 
-    public OasdiffManager(IProcessWrapper processWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
+    public OasdiffManager(ILoggerWrapper loggerWrapper, IProcessWrapper processWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
     {
+        this._loggerWrapper = loggerWrapper;
         this._httpClientWrapper = httpClientWrapper;
         this._processWrapper = processWrapper;
-        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
+        this._openApiToolsDirectory = openApiToolsDirectoryPath;
+        this._oasdiffDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
     }
 
+    private bool SkipRun { get; set; }
+    
     public async Task InstallOasdiffAsync(CancellationToken cancellationToken)
     {
-        Directory.CreateDirectory(this._toolDirectory);
+        try
+        {
+            this._loggerWrapper.LogMessage("Starting Oasdiff installation.");
+            
+            Directory.CreateDirectory(this._oasdiffDirectory);
+
+            var oasdiffFileName = GetOasdiffFileName();
+            var url = $"https://github.com/Tufin/oasdiff/releases/download/v{OasdiffVersion}/{oasdiffFileName}";
+
+            await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._oasdiffDirectory, oasdiffFileName), cancellationToken);
+            await this.DecompressDownloadedFileAsync(oasdiffFileName, cancellationToken);
```

Comment:
Could be optimized later on. Right now we skip the download if the tar.gz file already exists.

IMO, ideally the flow should be to skip the download AND decompression if the decompressed binary is already present.

Currently we always decompress the tar.gz file during each build